stats/src/test/java/com/facebook/airlift/stats/TestQuantileDigest.java

```diff
@@ -766,6 +766,91 @@ public void testMergeSameLevel()
     }
 
+    @Test
+    public void testScale()
+            throws Exception
+    {
+        QuantileDigest digest = new QuantileDigest(1);
+
+        addAll(digest, asList(0, 1, 2, 3, 4, 5, 6, 7, 8, 9));
+
+        // Should have no compressions with so few values and the allowed error.
+        assertEquals(digest.getConfidenceFactor(), 0.0);
+
+        // Before scaling.
+        assertEquals(digest.getHistogram(asList(0L, 1L, 2L, 3L, 4L, 5L, 6L, 7L, 8L, 9L, 10L)),
+                asList(new QuantileDigest.Bucket(0, Double.NaN),
+                        new QuantileDigest.Bucket(1, 0),
+                        new QuantileDigest.Bucket(1, 1),
+                        new QuantileDigest.Bucket(1, 2),
+                        new QuantileDigest.Bucket(1, 3),
+                        new QuantileDigest.Bucket(1, 4),
+                        new QuantileDigest.Bucket(1, 5),
+                        new QuantileDigest.Bucket(1, 6),
+                        new QuantileDigest.Bucket(1, 7),
+                        new QuantileDigest.Bucket(1, 8),
+                        new QuantileDigest.Bucket(1, 9)));
+
+        // Scale up the counts.
+        digest.scale(4);
+        assertEquals(digest.getHistogram(asList(0L, 1L, 2L, 3L, 4L, 5L, 6L, 7L, 8L, 9L, 10L)),
+                asList(new QuantileDigest.Bucket(0, Double.NaN),
+                        new QuantileDigest.Bucket(4, 0),
+                        new QuantileDigest.Bucket(4, 1),
+                        new QuantileDigest.Bucket(4, 2),
+                        new QuantileDigest.Bucket(4, 3),
+                        new QuantileDigest.Bucket(4, 4),
+                        new QuantileDigest.Bucket(4, 5),
+                        new QuantileDigest.Bucket(4, 6),
+                        new QuantileDigest.Bucket(4, 7),
+                        new QuantileDigest.Bucket(4, 8),
+                        new QuantileDigest.Bucket(4, 9)));
+
+        // Scale down the counts.
+        digest.scale(0.5);
+        assertEquals(digest.getHistogram(asList(0L, 1L, 2L, 3L, 4L, 5L, 6L, 7L, 8L, 9L, 10L)),
+                asList(new QuantileDigest.Bucket(0, Double.NaN),
+                        new QuantileDigest.Bucket(2, 0),
+                        new QuantileDigest.Bucket(2, 1),
+                        new QuantileDigest.Bucket(2, 2),
+                        new QuantileDigest.Bucket(2, 3),
+                        new QuantileDigest.Bucket(2, 4),
+                        new QuantileDigest.Bucket(2, 5),
+                        new QuantileDigest.Bucket(2, 6),
+                        new QuantileDigest.Bucket(2, 7),
+                        new QuantileDigest.Bucket(2, 8),
+                        new QuantileDigest.Bucket(2, 9)));
+    }
+
+    @Test
+    public void testScaleRemoveSmallCount()
+            throws Exception
+    {
+        QuantileDigest digest = new QuantileDigest(1);
+        digest.add(0);
+        for (int i = 0; i < 10e5; i++) {
```

Comment:
It would be more accurate to have this loop over some number of values (perhaps same values as above), and just have one/some of them have a weight of `10e5` (see `QuantileDigest#add(int, double)` for overload which takes in fractional weight)

Reply:
I don't follow you here. the idea of this test is to have 1 time of value "0" and 10e5 times of value "1".
if you mean to cover the fractional count (weight), I can do that in other test.

Reply:
Right, I was confused by the purpose of the loop.  You can just make this test `add(1, 10e5)` once.  Also, we should make sure we test the count is rescaled as well.  Other than that, it's fine.  It would also be good to show fractional weight behavior.

Reply:
Both tests show that the counts changed after scale (with fractional factor too). This test further show that if the new value is too close to zero, it cut out.